docs/web3-eth-miner.rst

```diff
@@ -0,0 +1,224 @@
+.. _eth-miner:
+
+.. include:: include_announcement.rst
+
+==============
+web3.eth.miner
+==============
+
+
+The ``web3-eth-miner`` package allows you to remote control the node's mining operation and set various mining specific settings.
+
+
+.. code-block:: javascript
+
+    import {Miner} from 'web3-eth-miner';
+
+    // "Web3.givenProvider" will be set if in an Ethereum supported browser.
+    const miner = new Miner(Web3.givenProvider || 'ws://some.local-or-remote.node:8546', null, options);
+
+
+------------------------------------------------------------------------------
+
+
+.. include:: include_package-core.rst
+
+
+
+------------------------------------------------------------------------------
+
+
+setExtra
+========
+
+.. code-block:: javascript
+
+    web3.eth.miner.setExtra(extraData, [, callback])
```

Comment:
this should be: ``miner.setExtra(extraData, [, callback])``

Reply:
can you explain @nivida. This module is web3-eth-miner so, you mean that this module should be changed to **web3-miner** module rather than **web3-eth-miner**?
This means that all methods will be called like **web3.miner.setExtra(extraData, [, callback])**

Reply:
The name can be called web3-eth-miner.. but the miner module shouldn't be a submodule of web3-eth. It should be a standalone module. The problem is that these methods aren't used that much and it will just blown up the bundle size.

Reply:
This module should only be used this way: 

``` javascript 
import {Miner} from 'web3-eth-miner';
const miner = new Miner(...);
miner.setExtra(...).then(console.log);

```

Reply:
This is a standalone module only @nivida. I didn't export this module to web3-eth. 
I will rename all the methods in docs as suggested.

Reply:
Also, I think in admin and txpool documentation, I have to change this.